tests/timeseries/test_base_types.py

```diff
@@ -0,0 +1,74 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for timeseries base types."""
+
+
+from datetime import datetime
+
+from frequenz.sdk.timeseries._base_types import Bounds, SystemBounds
+from frequenz.sdk.timeseries._quantities import Power
+
+
+def test_bounds_contains() -> None:
+    """Test the `__contains__` method of the `Bounds` class.
+
+    This method checks if a value is within the defined bounds.
+    """
+    bounds = Bounds(lower=Power.from_watts(10), upper=Power.from_watts(100))
+
+    assert Power.from_watts(50) in bounds  # within bounds
+    assert Power.from_watts(10) in bounds  # at lower bound
+    assert Power.from_watts(100) in bounds  # at upper bound
+    assert Power.from_watts(9) not in bounds  # below lower bound
+    assert Power.from_watts(101) not in bounds  # above upper bound
+
+    bounds_no_lower = Bounds(lower=None, upper=Power.from_watts(100))
+    assert Power.from_watts(50) in bounds_no_lower  # within upper bound
+    assert Power.from_watts(100) in bounds_no_lower  # at upper bound
+    assert Power.from_watts(101) not in bounds_no_lower  # above upper bound
+
+    bounds_no_upper = Bounds(lower=Power.from_watts(10), upper=None)
+    assert Power.from_watts(50) in bounds_no_upper  # within lower bound
+    assert Power.from_watts(10) in bounds_no_upper  # at lower bound
+    assert Power.from_watts(9) not in bounds_no_upper  # below lower bound
```

Comment:
Small suggestion, if you split this in a separate test, you will get better test reports if one case fails but the other don't. For example, if the test with lower and upper bounds fail, the test without a lower bound (and without an upper bound) will not be executed, so you will not know if they succeeded or not.

If you make a separate test for no upper bound and another for no lower bound, all 3 will be tested and you'll get separate results for each.